menpo3d/morphablemodel/fitter.py

```diff
@@ -0,0 +1,382 @@
+import sys
+import time
+import numpy as np
+import menpo.io as mio
+import menpo3d.io as m3dio
+import matplotlib.pyplot as plt
+from menpo.feature import gradient
+from menpo.image import Image
+from menpo3d.rasterize import GLRasterizer
+from menpo.transform import Homogeneous, UniformScale
+from .lmalign import retrieve_view_projection_transforms
+from .detectandfit import detect_and_fit
+from .derivatives import compute_pers_projection_derivatives_shape_parameters, \
+                              compute_ortho_projection_derivatives_shape_parameters, \
+                              compute_ortho_projection_derivatives_warp_parameters, \
+                              compute_pers_projection_derivatives_warp_parameters, \
+                              compute_texture_derivatives_texture_parameters, \
+                              compute_projection_derivatives_warp_parameters, \
+                              compute_projection_derivatives_shape_parameters
+
+
+class MMFitter(object):
+    r"""
+    Class for defining a 3DMM fitter.
+
+    """
+    def __init__(self, mm):
+        self._model = mm
+        self.result = None
+        self.rasterized_result = None
+        self.rasterizer = None
+        self.errors = []
+
+    @property
+    def mm(self):
+        r"""
+        The 3DMM model.
+
+        """
+        return self._model
+
+    @property
+    def get_result(self):
+        r"""
+        The 3DMM model fitting result.
+
+        """
+        return self.result
+
+    # TODO
+    def fit(self, image_path, n_alphas=100, n_betas=100, n_tris=1000, camera_update=False, max_iters=100):
```

Comment:
We may wish to use this fitter on an image that isn't on disk - e.g. one we may have acquired directly from a [webcam](https://github.com/menpo/menpowidgets/blob/master/menpowidgets/base.py#L2250).

Therefore, take a menpo image as input here, and update the notebook to be:
```py
image = mio.import_image(image_path)
fitter.fit(image, ...)
```